### ISSUE TITLE
Cleaner File Upload API

### ISSUE THREAD

**Issue by scottynoshotty** [781flyingdutchman/background_downloader#264]
The UploadTask API is incredibly confusing. Callers of the API need to understand how to map the path to the File they want to upload into some combination of base directory, directory and filename. Is there any reason why we can't just provide the path to the file? If there's potential for misunderstanding of a String filePath we could also just provide a File object.

**Comment by 781flyingdutchman** [comment]
Per the [documentation](https://pub.dev/packages/background_downloader#specifying-the-location-of-the-file-to-download-or-upload), there is a good reasons why passing an absolute path is not a good idea: the location of the app's documents directory may change between application starts (on iOS, and on Android in some cases), and uploads with an absolute path may therefore fail if the upload starts while the app is suspended or if it is restarted. Just look through the `flutter_downloader` issue list to see how many problems this causes - it was in fact the reason I decided to create an alternative to `flutter_downloader`. It's therefore not something I want to encourage, plus it would add more complexity to the API by providing an alternative. 

It's trivial to create the upload task based on a File if you are comfortable using an absolute path:
```
import 'package:path/path.dart' as p;  // for easy path manipulation

final file = File('something.txt');
final task = UploadTask(
    url: 'https://server.com',
    baseDirectory: BaseDirectory.root,
    directory: p.dirname(file.absolute.path),
    filename: p.basename(file.absolute.path));
```

**Comment by 781flyingdutchman** [comment]
Upon reflection, I decided to add `UploadTask.fromFile` to make this possible, albeit with warnings about its use. It's on the `dev` branch right now, so if you can test it please let me know if it works as intended.

**Comment by scottynoshotty** [comment]
Hey Flying Dutchman, I tested it out on iOS and Android and it appears to be working well. Thanks for the quick turnaround!

**Comment by 781flyingdutchman** [comment]
Cool, published as V8.3.0 and thanks for your support!  I added another method that may be helpful in this context.

If you already have a path to a file or a `File` object, you can extract the values for `baseDirectory`, `directory` and `filename` using `Task.split` to create the task:
```dart
final (baseDirectory, directory, filename) = await Task.split(filePath: yourPath);
final task = UploadTask(
        url: 'https://yourserver.com',
        baseDirectory: baseDirectory,
        directory: directory,
        filename: filename);
```

`Task.split` takes a filePath or a file as argument, and IMHO this is a better way to create your `UploadTask` than using `UploadTask.fromFile`, as it figures out the appropriate baseDirectory/directory/filename convention (instead of just using `BaseDirectory.root`) and is therefore safer on mobile platforms.

**Comment by danieljiwonkang98** [comment]
@781flyingdutchman UploadTask.fromFile can't be seen in the 7.11.0 version
